### ISSUE TITLE
[BUG] Stack Overflow when run `metrics .`

### ISSUE THREAD

**Issue by lsaudon** [dart-code-checker/dart-code-metrics#1087]
* **Dart code metrics version:** 5.1.0
* **Dart sdk version:** 2.18.5

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
name: my_app
description: A Very Good Project created by Very Good CLI.
version: 1.0.0+1
publish_to: none

environment:
  sdk: "2.18.5"
  flutter: "3.3.9"

dependencies:
  bloc: ^8.1.0
  equatable: ^2.0.5
  flutter:
    sdk: flutter
  flutter_bloc: ^8.1.1
  flutter_localizations:
    sdk: flutter
  flutter_markdown: ^0.6.13
  flutter_native_splash: ^2.2.16
  go_router: ^5.2.0
  http: ^0.13.5
  intl: ^0.17.0
  package_info_plus: ^3.0.2
  provider: ^6.0.4
  shared_preferences: ^2.0.15
  url_launcher: ^6.1.7

dev_dependencies:
  bloc_test: ^9.1.0
  faker: ^2.0.0
  flutter_test:
    sdk: flutter
  integration_test:
    sdk: flutter
  mocktail: ^0.3.0
  my_lints:
    path: ../my_lints

dependency_overrides:
  analyzer: ^5.2.0 # Check when `flutter` or `mocktail` is updated if it is still useful.

flutter:
  uses-material-design: true
  generate: true
```

```yaml
name: my_lints
description: Lints for Flutter apps, packages, and plugins.
version: 0.1.0+1
publish_to: none

environment:
  sdk: "2.18.5"

dependencies:
  dart_code_metrics: ^5.1.0
```

</details>

**What did you expect to happen?**

No exception

**What actually happened?**

```sh
⠇ Analyzing 104 file(s)... 3.7sUnhandled exception:
Stack Overflow
#0      StringBuffer._addPart (dart:core-patch/string_buffer_patch.dart:160:3)
#1      StringBuffer.write (dart:core-patch/string_buffer_patch.dart:67:5)
#2      ToSourceVisitor.visitSimpleIdentifier (package:analyzer/src/dart/ast/to_source_visitor.dart:1170:10)
#3      SimpleIdentifierImpl.accept (package:analyzer/src/dart/ast/ast.dart:11474:50)
#4      ToSourceVisitor.visitNamedType (package:analyzer/src/dart/ast/to_source_visitor.dart:889:5)
#5      NamedTypeImpl.accept (package:analyzer/src/dart/ast/ast.dart:9017:50)
#6      ToSourceVisitor._visitNodeList (package:analyzer/src/dart/ast/to_source_visitor.dart:1433:18)
#7      ToSourceVisitor.visitTypeArgumentList (package:analyzer/src/dart/ast/to_source_visitor.dart:1313:5)
#8      TypeArgumentListImpl.accept (package:analyzer/src/dart/ast/ast.dart:13055:50)
#9      ToSourceVisitor.visitNamedType (package:analyzer/src/dart/ast/to_source_visitor.dart:890:5)
#10     NamedTypeImpl.accept (package:analyzer/src/dart/ast/ast.dart:9017:50)
#11     ToSourceVisitor.visitTypeLiteral (package:analyzer/src/dart/ast/to_source_visitor.dart:1319:5)
#12     TypeLiteralImpl.accept (package:analyzer/src/dart/ast/ast.dart:13153:50)
#13     ToSourceVisitor._visitNodeList (package:analyzer/src/dart/ast/to_source_visitor.dart:1433:18)
#14     ToSourceVisitor.visitArgumentList (package:analyzer/src/dart/ast/to_source_visitor.dart:38:5)
#15     ArgumentListImpl.accept (package:analyzer/src/dart/ast/ast.dart:396:50)
#16     ToSourceVisitor.visitMethodInvocation (package:analyzer/src/dart/ast/to_source_visitor.dart:865:5)
#17     MethodInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:8704:50)
#18     ToSourceVisitor._visitNodeList (package:analyzer/src/dart/ast/to_source_visitor.dart:1433:18)
#19     ToSourceVisitor.visitArgumentList (package:analyzer/src/dart/ast/to_source_visitor.dart:38:5)
#20     ArgumentListImpl.accept (package:analyzer/src/dart/ast/ast.dart:396:50)
#21     ToSourceVisitor.visitMethodInvocation (package:analyzer/src/dart/ast/to_source_visitor.dart:865:5)
#22     AstNodeImpl.toSource (package:analyzer/src/dart/ast/ast.dart:877:5)
#23     AstNodeImpl.toString (package:analyzer/src/dart/ast/ast.dart:882:24)
#24     _StringBase._interpolate (dart:core-patch/string_patch.dart:853:19)
#25     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:65:50)
#26     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:59:7)
#27     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:67:7)
#28     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:59:7)
#29     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:67:7)
#30     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:59:7)
#31     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:67:7)
#32     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:59:7)
#33     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:67:7)
#34     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:59:7)
#35     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:67:7)
#36     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:59:7)
#37     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:67:7)
#38     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:59:7)
#39     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:67:7)
#40     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:59:7)
#41     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:67:7)
#42     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:59:7)
#43     _Visitor._traverseParents (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:67:7)
...
...
#14859  GeneralizingAstVisitor.visitFunctionExpression (package:analyzer/dart/ast/visitor.dart:380:58)
#14860  FunctionExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:5952:50)
#14861  FunctionDeclarationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:5824:25)
#14862  GeneralizingAstVisitor.visitNode (package:analyzer/dart/ast/visitor.dart:518:10)
#14863  GeneralizingAstVisitor.visitAnnotatedNode (package:analyzer/dart/ast/visitor.dart:135:48)
#14864  GeneralizingAstVisitor.visitDeclaration (package:analyzer/dart/ast/visitor.dart:257:44)
#14865  _Visitor.visitDeclaration (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/visitor.dart:40:11)
#14866  GeneralizingAstVisitor.visitCompilationUnitMember (package:analyzer/dart/ast/visitor.dart:219:7)
#14867  GeneralizingAstVisitor.visitNamedCompilationUnitMember (package:analyzer/dart/ast/visitor.dart:499:7)
#14868  GeneralizingAstVisitor.visitFunctionDeclaration (package:analyzer/dart/ast/visitor.dart:372:12)
#14869  FunctionDeclarationImpl.accept (package:analyzer/src/dart/ast/ast.dart:5818:50)
#14870  NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:9237:20)
#14871  CompilationUnitImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:2482:21)
#14872  BanNameRule.check (package:dart_code_metrics/src/analyzers/lint_analyzer/rules/rules_list/ban_name/ban_name_rule.dart:43:17)
#14873  LintAnalyzer._checkOnCodeIssues.<anonymous closure> (package:dart_code_metrics/src/analyzers/lint_analyzer/lint_analyzer.dart:278:22)
#14874  ExpandIterator.moveNext (dart:_internal/iterable.dart:481:31)
#14875  new _GrowableList._ofOther (dart:core-patch/growable_array.dart:202:26)
#14876  new _GrowableList.of (dart:core-patch/growable_array.dart:152:26)
#14877  Iterable.toList (dart:core/iterable.dart:470:12)
#14878  LintAnalyzer._checkOnCodeIssues (package:dart_code_metrics/src/analyzers/lint_analyzer/lint_analyzer.dart:283:12)
#14879  LintAnalyzer._analyzeFile (package:dart_code_metrics/src/analyzers/lint_analyzer/lint_analyzer.dart:220:21)
#14880  LintAnalyzer.runCliAnalysis (package:dart_code_metrics/src/analyzers/lint_analyzer/lint_analyzer.dart:123:26)
#14881  _RootZone.runUnary (dart:async/zone.dart:1653:54)
#14882  _FutureListener.handleValue (dart:async/future_impl.dart:147:18)
#14883  Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:766:45)
#14884  Future._propagateToListeners (dart:async/future_impl.dart:795:13)
#14885  Future._completeWithValue (dart:async/future_impl.dart:566:5)
#14886  _RootZone.runUnary (dart:async/zone.dart:1653:54)
#14887  _FutureListener.handleValue (dart:async/future_impl.dart:147:18)
#14888  Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:766:45)
#14889  Future._propagateToListeners (dart:async/future_impl.dart:795:13)
#14890  Future._completeWithValue (dart:async/future_impl.dart:566:5)
#14891  Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:639:7)
#14892  _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#14893  _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
#14894  _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:122:13)
#14895  _Timer._runTimers (dart:isolate-patch/timer_impl.dart:405:11)
#14896  _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
#14897  _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:192:12)
```
**Are you willing to submit a pull request to fix this bug?**

No

**Comment by lsaudon** [comment]
I try with `dart_code_metrics 5.0.1` on dart pub global.
I don't have a Stack Overflow error

**Comment by incendial** [comment]
Fixed in 5.2.0 🚀
